solidity/contracts/vortex/VortexBurner.sol

```diff
@@ -0,0 +1,522 @@
+     */
+    function transferNetworkFeeWalletOwnership(address newOwner) external ownerOnly {
+        networkFeeWallet().transferOwnership(newOwner);
+    }
+
+    /**
+     * @dev accepts the ownership of he network fee wallet
+     */
+    function acceptNetworkFeeOwnership() external ownerOnly {
+        networkFeeWallet().acceptOwnership();
+    }
+
+    /**
+     * @dev returns the vortex conversion strategy for the specified tokens
+     *
+     * @param tokens the tokens to convert
+     *
+     * @return the conversion paths for each of tokens to convert
+     * @return the conversion amounts for each of tokens to convert
+     * @return the conversion path for the last network to governance tokens conversion
+     */
+    function vortexStrategy(IERC20[] calldata tokens, ITokenHolder feeWallet)
+        private
+        view
+        returns (
+            address[][] memory,
+            uint256[] memory,
+            address[] memory
+        )
+    {
+        IConverterRegistry registry = converterRegistry();
+
+        // create conversion paths and collect available token amounts
+        address[][] memory paths = new address[][](tokens.length);
+        uint256[] memory amounts = new uint256[](tokens.length);
+
+        for (uint256 i = 0; i < tokens.length; ++i) {
+            IERC20 token = tokens[i];
+
+            address[] memory path = new address[](3);
+            path[0] = address(token);
+
+            // don't look up for a converter for either the network or the governance token, since they are going to be
+            // handled in a special way during the burn itself
+            if (token != _networkToken && token != _govToken) {
+                path[1] = address(networkTokenConverterAnchor(token, registry));
+                path[2] = address(_networkToken);
+            }
+
+            paths[i] = path;
+
+            // make sure to retrieve the balance of either an ERC20 or an ETH reserve
+            if (token == NATIVE_TOKEN_ADDRESS) {
+                amounts[i] = address(feeWallet).balance;
+            } else {
+                amounts[i] = token.balanceOf(address(feeWallet));
+            }
+        }
+
+        // get the governance token converter path
+        address[] memory govPath = new address[](3);
+        govPath[0] = address(_networkToken);
+        govPath[1] = address(networkTokenConverterAnchor(_govToken, registry));
+        govPath[2] = address(_govToken);
+
+        return (paths, amounts, govPath);
+    }
+
+    /**
+     * @dev applies the burn incentive fee on the provided amount and returns the net amount and the fee
+     *
+     * @param amount the network tokens amount
+     *
+     * @return the net amount
+     * @return the fee amount
+     */
+    function applyIncentiveFee(uint256 amount) private view returns (uint256, uint256) {
+        uint256 fee = _burnIncentiveFee;
+        if (fee == 0) {
+            return (amount, 0);
```

Comment:
This piece of code (special case) is redundant.
You can reduce this function to be only the last two lines of it.

Reply:
I wanted to avoid the extra gas costs in this case

Reply:
It's an extremely minor gas-cost (a couple of arithmetic operations on non-storage variables), and in any case, the branch itself also yields some gas-cost.
IMO, when it comes to minor gas issues like this one vs an additional piece code which is effectively redundant, I'd opt for cleaner code.

Reply:
I don't mind either way. I'll change it